src/kem/kyber/libjade_kyber512_avx2/api.c

```diff
@@ -0,0 +1,19 @@
+#include <oqs/rand.h>
+#include "api.h"
+
+int libjade_kyber512_avx2_keypair(uint8_t *public_key, uint8_t *secret_key) {
+    uint8_t keypair_coins[JADE_KEM_kyber_kyber512_amd64_avx2_KEYPAIRCOINBYTES];
+    OQS_randombytes(keypair_coins, JADE_KEM_kyber_kyber512_amd64_avx2_KEYPAIRCOINBYTES/2);
```

Comment:
Quick sanity check for my own understanding: I'm assuming we can't get the randomness in one shot because that would screw up the KATs and their counter-based PRNG?

Reply:
Yes, precisely for the reason you mention.